test/e2e/pods/pods.go

```diff
@@ -50,18 +56,259 @@ func ExecCommand(cs *testclient.ClientSet, pod *corev1.Pod, command ...string) (
+		return pingV6(client, sourcePod, targetIP)
+	default:
+		panic("Unexpected protocol")
+	}
+}
+
+func pingV4(client *testclient.ClientSet, sourcePod *corev1.Pod, targetIP string) (string, string, error) {
+	return ping(client, sourcePod, targetIP, false)
+}
+
+func pingV6(client *testclient.ClientSet, sourcePod *corev1.Pod, targetIP string) (string, string, error) {
+	return ping(client, sourcePod, targetIP, true)
+}
+
+func ping(client *testclient.ClientSet, sourcePod *corev1.Pod, targetIP string, isV6 bool) (string, string, error) {
+	if isV6 {
+		return ExecCommand(client, sourcePod, "ping", "-6", "-c", "1", targetIP)
+	}
+	return ExecCommand(client, sourcePod, "ping", "-4", "-c", "1", targetIP)
+}
+
+func ConnectToPortFromPod(client *testclient.ClientSet, proto ingressnodefwiov1alpha1.IngressNodeFirewallRuleProtocolType, v6 bool,
+	sourcePod *corev1.Pod, ip, dPort string) (string, string, error) {
+	switch proto {
+	case ingressnodefwiov1alpha1.ProtocolTypeTCP:
+		if v6 {
+			return iperfTCPV6(client, sourcePod, ip, dPort)
+		}
+		return iperfTCPV4(client, sourcePod, ip, dPort)
+	case ingressnodefwiov1alpha1.ProtocolTypeUDP:
+		if v6 {
+			return iperfUDPV6(client, sourcePod, ip, dPort)
+		}
+		return iperfUDPV4(client, sourcePod, ip, dPort)
+	case ingressnodefwiov1alpha1.ProtocolTypeSCTP:
+		if v6 {
+			return iperfSCTPV6(client, sourcePod, ip, dPort)
+		}
+		return iperfSCTPV4(client, sourcePod, ip, dPort)
+	default:
+		panic("Unsupported protocol")
+	}
+}
+
+func iperfTCPV4(client *testclient.ClientSet, sourcePod *corev1.Pod, ip, dPort string) (string, string, error) {
+	return iperfTransport(client, sourcePod, ip, dPort, "-4")
+}
+
+func iperfTCPV6(client *testclient.ClientSet, sourcePod *corev1.Pod, ip, dPort string) (string, string, error) {
+	return iperfTransport(client, sourcePod, ip, dPort, "-6")
+}
+
+func iperfUDPV4(client *testclient.ClientSet, sourcePod *corev1.Pod, ip, dPort string) (string, string, error) {
+	return iperfTransport(client, sourcePod, ip, dPort, "--udp", "-4")
+}
+
+func iperfUDPV6(client *testclient.ClientSet, sourcePod *corev1.Pod, ip, dPort string) (string, string, error) {
+	return iperfTransport(client, sourcePod, ip, dPort, "--udp", "-6")
+}
+
+func iperfSCTPV4(client *testclient.ClientSet, sourcePod *corev1.Pod, ip, dPort string) (string, string, error) {
+	return iperfTransport(client, sourcePod, ip, dPort, "--sctp", "-4")
+}
+
+func iperfSCTPV6(client *testclient.ClientSet, sourcePod *corev1.Pod, ip, dPort string) (string, string, error) {
+	return iperfTransport(client, sourcePod, ip, dPort, "--sctp", "-6")
+}
+
+func iperfTransport(client *testclient.ClientSet, sourcePod *corev1.Pod, ip, dPort string, additionalFlag ...string) (string, string, error) {
+	command := []string{"iperf3", "--connect-timeout", "1000", "--time", "1", "--client", ip, "--port", dPort}
+	command = append(command, additionalFlag...)
+	return ExecCommand(client, sourcePod, command...)
```

Comment:
nit wouldn't be better to move iperf and ping methods to new pkg and we just keep this pkg for pod's related methods ?